### DISCUSSION TITLE
Where is the config stored?

### DISCUSSION THREAD

**Discussion by Destroy666x** [nefarius/HidHide#130] [Q&A] [answered]
Where is it stored and can it be backed up -> restored in working state, assuming the same controllers would be used on another device?

**Comment by nefarius** [comment, marked as answer]
Outside of user's reach since they tend to tinker with stuff and break things 😉 The only supported way to fetch and alter configs is via the driver API.

If you still want to know; the sources reveal this "secret" 😉

**Reply by Destroy666x** [reply]
Ah ok, thanks for info. I guess I'll just skip it in my upcoming app backup software then as I'm not a big fan of software that thinks users are "danger" (well, they are, but not to the point of such implementations).

**Reply by nefarius** [reply]
Trust me, I am in this business long enough to know many users are their own biggest threat 😉 But to answer your question seriously; it is in the registry, so whenever you make a system backup that includes it, you have your settings backed up too.

You can find it here `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\HidHide\Parameters` but it is read/write protected by default to not repeat the mistakes I made with HidGuardian 😜

**Reply by Destroy666x** [reply]
The point of the software is to avoid doing full system/registry/whatever backups. You don't always want that.

I see, that's why they couldn't be found even by full path of whitelisted apps. Thanks for further info.

**Reply by nefarius** [reply]
> The point of the software is to avoid doing full system/registry/whatever backups. You don't always want that.

I see. Unfortunately it is not possible to make kernel driver configuration file-based, we have to live with the registry.
